File: lib/data/non/input/mouse.rb
module Mouse
    java_import 'com.badlogic.gdx.Gdx'
    java_import 'non.Non'
    
    def self.show
        self.setVisible(true)
    end
    
    def self.hide
        self.setVisible(false)
    end
    
    def self.x
        Gdx.input.getX()
    end
    
    def self.y
        Gdx.input.getY()
    end
    
    def self.move(x, y)
        Gdx.input.setCursorPosition(x, y)
    end
    
    def self.down(button = :left)
        buttoncode = Non.getButton(name)
        Gdx.input.isButtonPressed(buttoncode)
    end
end

File: lib/data/non/lua/input/mouse.lua
non.mouse = {}

local Gdx = NON_GDX
local Non = NON_NON
  
function non.mouse.show()
  Gdx.input:setCursorCatched(false)
end
  
function non.mouse.hide()
  Gdx.input:setCursorCatched(true)
end
  
function non.mouse.x()
  return Gdx.input:getX()
end
  
function non.mouse.y()
  return Gdx.input:getY()
end
  
function non.mouse.move(x, y)
  return Gdx.input:setCursorPosition(x, y)
end
  
function non.mouse.down(button)
  if button == nil then buton = "left" end
  local buttoncode = Non:getButton(name)
  return Gdx.input:isButtonPressed(buttoncode)
end